x11-apps/mesa-progs/mesa-progs-8.3.0.ebuild

```diff
@@ -15,12 +15,12 @@ fi
 inherit base autotools toolchain-funcs ${GIT_ECLASS}
 
 DESCRIPTION="Mesa's OpenGL utility and demo programs (glxgears and glxinfo)"
-HOMEPAGE="http://mesa3d.sourceforge.net/"
+HOMEPAGE="https://www.mesa3d.org/ https://mesa.freedesktop.org/"
 if [[ ${PV} == 9999* ]]; then
 	SRC_URI=""
 	KEYWORDS="alpha amd64 arm hppa ia64 ppc ppc64 sparc x86"
 else
-	SRC_URI="ftp://ftp.freedesktop.org/pub/${MY_PN/-//}/${PV}/${MY_P}.tar.bz2"
+	SRC_URI="https://mesa.freedesktop.org/archive/${MY_PN/-//}/${PV}/${MY_P}.tar.bz2"
```

Comment:
It's reasonable to use P/PV/MY_P/... in SRC_URI because the version number changes. All other cases are pure abuse.